railties/lib/rails/commands/commands_tasks.rb

```diff
@@ -103,6 +110,18 @@ def new
       end
     end
 
+    def rake
+      command = parse_command(ARGV.shift)
+
+      case command
+        when 'help' then ARGV.unshift('--help')
+        when 'version' then ARGV.unshift('--version')
+        else  ARGV.unshift(command)
+      end
```

Comment:
I totally removed querying for help/version from the rake, as users will be able to find that info directly using `rake`.